### PR TITLE
[ONL-7409] Migrate ec-btn to TS

### PR DESCRIPTION
**Pull request by wolandec**

### CODE REVIEW

File: src/components/ec-btn/ec-btn.vue [hunk 1]
```diff
@@ -50,150 +50,115 @@
...
-    'ec-btn--full-width': isFullWidth.value,
-    'ec-btn--icon-only': icon.value && !hasDefaultSlot(),
-    [`ec-btn--${category.value}`]: category.value,
-    'ec-btn--outline': isOutline.value,
-    'ec-btn--primary-reverse': isReverse.value && category.value === 'primary',
-    'ec-btn--secondary-reverse': isReverse.value && category.value === 'secondary',
-    'ec-btn--success-reverse': isReverse.value && category.value === 'success',
-    'ec-btn--error-reverse': isReverse.value && category.value === 'error',
-    'ec-btn--warning-reverse': isReverse.value && category.value === 'warning',
+    'ec-btn--sm': props.size === 'sm',
```
Comment (stavros-tomas) [inline src/components/ec-btn/ec-btn.vue:150]: Shall we use the enums here? `ButtonSize.Sm`, and few lines below `ButtonCategory.Primary` etc?

Comment (wolandec) [reply src/components/ec-btn/ec-btn.vue:150]: Thank you! Sure we must!

File: src/components/ec-btn/ec-btn.spec.ts [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
+        props,
+        ...mountOpts,
+      },
+    ) as CVueWrapper;
   }
 
   it('should render a <button> element by default', () => {
-    const wrapper = mountBtn();
+    const wrapper = mountBtn({
+      category: ButtonCategory.Primary,
```
Comment (mcibique) [inline src/components/ec-btn/ec-btn.spec.ts:24]: We shouldn't change the tests while doing migrations. Why is this needed?

File: src/components/ec-btn/types.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import type { RouteLocationRaw } from 'vue-router';
+
+import type { IconName } from '../ec-icon/iconNames';
+
+export enum ButtonSize {
+  Md = 'md',
```
Comment (mcibique) [inline src/components/ec-btn/types.ts:6]: Enum names can use human readable string, rename it to Medium and Small. Keep the values in short form.

File: src/components/ec-btn/ec-btn.spec.ts (hunk 2 above)
Comment (wolandec) [reply src/components/ec-btn/ec-btn.spec.ts:24]: Because category is a required prop.

Comment (mcibique) [reply src/components/ec-btn/ec-btn.spec.ts:24]: It wasn't required before, we should fix that then: https://github.com/Ebury/chameleon/blob/master/src/components/ec-btn/ec-btn.vue#L108-L113

Comment (wolandec) [reply src/components/ec-btn/ec-btn.spec.ts:24]: My bad, I'll fix it. Thanks!

**Review by stavros-tomas — APPROVED** [review]

**Review by derrick-g — APPROVED** [review]
